tools/src/wyvern/target/corewyvernIL/expression/StringLiteral.java

```diff
@@ -28,10 +29,46 @@ public void setValue(java.lang.String value) {
 		this.value = value;
 	}
 
+    @Override
+    public int hashCode() {
+        return this.value.hashCode();
+    }
+
+    public boolean equals(String s) {
```

Comment:
Overloading equals() for multiple argument types is a bad practice in Java...it ends up being very fragile because overloading is based on static types, not dynamic types.  There should be just one equals() method in the class that takes an Object as an argument.